### PR TITLE
PMM-4192 MongoDB EXPLAIN JSON APIs.

### PR DESCRIPTION
**Pull request by percona-csalguero**
Added structs and methods for MongoDB explain
-[ ] FB https://github.com/Percona-Lab/pmm-submodules/pull/757

### CODE REVIEW

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-420/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-420/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-420/

File: api/agentpb/agent.proto [hunk 1]
```diff
@@ -133,6 +133,12 @@ message StartActionRequest {
     string dsn = 1;
     string table = 2;
   }
+  // MongoDBExplainParams describes MongoDB EXPLAIN action parameters.
+  message MongoDBExplainParams {
+    string dsn = 1;
+    string database = 2;
```
Comment (AlekSi) [inline api/agentpb/agent.proto]: Can it be a part of DSN?

File: api/managementpb/actions.proto [hunk 2]
```diff
@@ -309,6 +310,32 @@ message StartPostgreSQLShowIndexActionResponse {
...
+message StartMongoDBExplainActionRequest {
+  // pmm-agent ID where to run this Action.
+  string pmm_agent_id = 1;
+  // Service ID for this Action. Required.
+  string service_id = 2 [
+    (validator.field) = {
+      string_not_empty: true
+    }
+  ];
+  // SQL query. Required.
```
Comment (AlekSi) [inline api/managementpb/actions.proto]: Not SQL :)

File: api/managementpb/actions.proto [hunk 3]
```diff
@@ -399,6 +426,13 @@ service Actions {
       body: "*"
     };
   }
+  // StartMongoDBExplainAction starts PostgreSQL EXPLAIN Action.
```
Comment (AlekSi) [inline api/managementpb/actions.proto]: Not PostgreSQL.

File: api/agentpb/agent.proto [hunk 4]
```diff
@@ -133,6 +133,12 @@ message StartActionRequest {
...
+    string query = 3;
+  }
```
Comment (BupycHuk) [inline api/agentpb/agent.proto]: Is the json is the only format that we are going to implement?
If not maybe we should add output_format here?

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-420/

File: api/agentpb/agent.proto (hunk 1 above)
Comment (percona-csalguero) [reply api/agentpb/agent.proto]: After researching it I found that the `db` part of the dsn is not used as the default db but as the default auth db:
https://github.com/percona/pmm-agent/blob/master/vendor/go.mongodb.org/mongo-driver/x/network/connstring/connstring.go#L253
Example from the command line: 
```
mongo --username=root --password=root-password 127.0.0.1:27017/admin
MongoDB shell version v3.6.8
connecting to: mongodb://127.0.0.1:27017/admin
Implicit session: session { "id" : UUID("77e42c9f-9287-4b90-b9e2-0197b97755da") }
MongoDB server version: 4.0.10-5
```
but if you specify a different db (it already exists in my test)
```
mongo --username=root --password=root-password 127.0.0.1:27017/test
MongoDB shell version v3.6.8
connecting to: mongodb://127.0.0.1:27017/test
Implicit session: session { "id" : UUID("37d249fc-908e-450d-8adc-af70c48c18c0") }
MongoDB server version: 4.0.10-5
WARNING: shell and server versions do not match
2019-09-09T16:33:54.432-0300 E QUERY    [thread1] Error: Authentication failed. :
DB.prototype._authOrThrow@src/mongo/shell/db.js:1608:20
@(auth):6:1
@(auth):1:2
exception: login failed
```
So, we do need the Database parameter

File: api/agentpb/agent.proto (hunk 4 above)
Comment (percona-csalguero) [reply api/agentpb/agent.proto]: I thought about that but in contrast to MySQL, MongoDB only handle JSONs.
I don't think we want to implement a custom format.

**Comment by AlekSi** [comment]
We should merge it only when the tree is open for 2.5

**Comment by AlekSi** [comment]
@BupycHuk I think that one should be merged now – otherwise, we will have to update all other repos on Ready for Merge step.

**Pull request merged**